Workplace "Accidents"

Ikarrus opened the pull request:
Re: https://www.youtube.com/watch?v=llQL9RM_KOo
- Slipping while carrying items in your hand may cause ~accidents~
  - 30% chance to throw it at yourself
  - 30% chance to throw it onto the floor under you
  - 30% chance to throw the item towards the direction you're facing
  - CLUMSY people have chance of attacking themselves with the item
  - NODROP items cannot cause accidents
- Throwing reagent containers such as drinks or beakers will spill some of their contents onto whatever they hit
  - Reagent containers thrown by bartenders will never spill if it doesn't collide into anything.
- Added logging for slipping

Ikarrus committed:
Workplace "Accidents"

- Slipping while carrying items in your hand may cause ~accidents~
  - 30% chance to attack yourself
  - 30% chance to attack the floor
  - 30% chance to throw the item towards the direction you're facing
- Throwing reagent containers such as drinks or beakers may spill their contents onto whatever they hit

Ikarrus committed:
Squashed Commits

Chocobro commented:
this sounds rad

Anonus commented:
PUT SIGNS DOWN, JANITOOOOOOOOR

fj45 commented:
O-H&S in space!

Iamgoofball commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -507,3 +507,23 @@ var/const/GALOSHES_DONT_HELP = 8
 	if(head && (head.flags & HEADBANGPROTECT))
 		return 1
+
+/mob/living/carbon/proc/accident(var/obj/item/Item)
```
>item is uppercase

Ikarrus replied:
Yes, and?

RemieRichards replied:
It should just be I, not Item

Ikarrus replied:
I've actually been asked to avoid using them a they're not very descriptive

RemieRichards replied:
Whoever told you that is stupid.
It's always been an acceptable part of our standards for a variable to be the capital form of the first letter of the last segment of the path.

eg:
```var/obj/item/I```
<B>i</B>tem
<B>I</B>

Iamgoofball commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -507,3 +507,23 @@ var/const/GALOSHES_DONT_HELP = 8
+
+	switch(rand(1,100)) //Nothing special happens 91-100
+		if(1 to 30) //attack yourself
+			src.Click()
```
>using src.

xxalpha commented:
:+1: for throwing acid beakers at someone

AlexanderUlanH commented:
:+1: to the slipping changes
:-1: to a proc called impact_reagent having a 5% chance to do nothing
:-1: to it being added to /obj/item and then only being called by certain reagent containers' throw_impact
Proc looks ugly, but most of that's logging.
Makes bottle slinging weaker than in #9778, but it was probably a bit too strong anyway.

Ikarrus commented:
@Alek2ander Thanks, I had no idea is_open_container() was a thing.

And yeah the 5% chance thing was probably a bad idea. I was imagining the bartender sliding drinks down the table, but that was a horrible way to do it.

RemieRichards commented on code/game/turfs/turf.dm:
```diff
@@ -228,6 +228,19 @@
+			playsound(M.loc, 'sound/misc/slip.ogg', 50, 1, -3)
+
+			var/active_hand = M.hand
+			M.hand = 1
```
explain.
you shouldn't have to force these to use accident()

PKPenguin321 commented:
\>inb4 a clown slips someone as an innocent joke and they hit themselves into crit and the clown gets banned

GunHog commented:
This is a good feature, but you SERIOUSLY need to talk with the other admins and found out policy-wise how they will deal with issues. This WILL cause problems with clowns and janitors, and we need to avoid having unwarranted bans as a result.

Ikarrus commented:
As an admin I don't see any problems?

Niknakflak commented:
Now people have reasons to not run with knives in their hand.

duncathan commented:
This buffs syndie soap. I love it.

Alek2ander commented:
Feature request: scissors. Don't run with scissors.

fj45 commented:
Urist Mc Station already has scissors.

Miauw62 commented:
as somebody who only plays janitor anymore, :+1: x :100:

Tokiko1 commented:
If you slip while holding sharp objects and throw them at yourself, can they embed?

vista- commented:
I cannot wait to see double e-sword murderboners slipping on banana peels and impaling themselves!

Ricotez commented:
:+1: to make harmless clown pranks go horribly wrong

Kromgar commented:
Is adding furries a harmless clown prank gone horribly wrong?

Ricotez commented:
please keep all shit flung at me limited to the feedback thread and the PR

Dorsisdwarf commented:
:+1:

AlexCrimson commented:
If we combine this with dismemberment can we have traitors slipping and decapping themselves with their own double-bladed Esword?

Ricotez commented:
I can definitely add support for that, I wrote a nice proc for all items that takes a target organ and decides if said organ is dismembered by the item or not

Jalleo commented:
If that happens the new powergamer will be stocked with lube.

Ricotez commented:
it wouldn't be any more unbalanced than dismemberment in the first place, because your limb needs to pass the item's damage threshold before it can dismember it, and only for eswords the threshold is 0

Shapsy commented:
I can't wait for a clown to try and slip a roboticist carrying a beaker of sulphuric acid only to get all their shit melted

AlexCrimson commented:
You know, Clown specifically should have a higher chance of hitting themselves when they slip.

Could also buff tator soap to always cause the person to hit themselves when slipped by it.

Ikarrus commented:
Read the PR description

Sent from my iPhone

> On Jun 29, 2015, at 12:53 PM, AlexCrimson <notifications@github.com> wrote:
> 
> You know, Clown specifically should have a higher chance of hitting themselves when they slip.
> 
> Could also buff tator soap to always cause the person to hit themselves when slipped by it.
> 
> —
> Reply to this email directly or view it on GitHub.
>

Ikarrus commented:
Do you even keep up with updates? Read the PR desc

Sent from my iPhone

> On Jun 29, 2015, at 12:53 PM, AlexCrimson <notifications@github.com> wrote:
> 
> You know, Clown specifically should have a higher chance of hitting themselves when they slip.
> 
> Could also buff tator soap to always cause the person to hit themselves when slipped by it.
> 
> —
> Reply to this email directly or view it on GitHub.
>

phil235 commented on code/modules/food&drinks/drinks/drinks/drinkingglass.dm:
```diff
@@ -8,6 +8,7 @@
 	volume = 50
 	burn_state = 0 //Burnable
 	burntime = 5
+	flags = OPENCONTAINER
```
Technically not needed since  /obj/item/weapon/reagent_containers/food/drinks already has it.

Lumipharon commented:
All looks good, but does this mean that as a bartender I can't throw drinks down the bar to people?
Because that would be pretty gay.

Ikarrus commented:
I will investigate

kevinz000 commented:
Uhh wouldn't this make chemists with acid beakers op since they just throw em at people and they get their shit melted?

YotaXP commented:
What is meant by "throw it onto the floor under you"?  Just drop?  I assume the remaining 10% chance is just drop.

Also, I find it odd that clumsy people are actually far less likely to throw the item forward.  How about swapping 'throw forward' and 'throw downward'?

PKPenguin321 commented:
>Uhh wouldn't this make chemists with acid beakers op since they just throw em at people and they get their shit melted?

Ikarrus committed:
- Swaps floor throwing and distance throwing
- Reagent containers thrown by bartenders will never spill if it doesn't collide into anything

Ikarrus commented:
Added a bartending skill.

Ikarrus commented:
>Uhh wouldn't this make chemists with acid beakers op since they just throw em at people and they get their shit melted?

Not that OP considering chemists can already walk up to you and do that in a number of more reliable ways.

>What is meant by "throw it onto the floor under you"? Just drop? I assume the remaining 10% chance is just drop.
Reagent containers spill when thrown. I've swapped them anyways as you suggested.

kevinz000 commented:
Oh come on that's like suggesting to make stunbatons and similar stun if you're hit when someone throws it at you..
I'm not giving any ideas.

PKPenguin321 commented:
yeah, it's giving you a direct ranged option which is really viable in a heated combat scenario, that is really strong

Lumipharon commented:
>only grandmaster elite level barmen can slide a drink down the bar without flipping it upside down and spilling it everywhere like a spastic

Oh well.

SconesC commented:
Janitors are such pieces of shit and we have so many summer players who just throw fucking buckets of water on the ground

God why, slips are already obscenely robust

Dorsisdwarf commented:
Because it's hilarious and isn't actually that big a change, except for spilling your drink / beaker of potassium 

The chance to hit yourself could be a taaad lower, maybe 15% instead of 30%

Ikarrus committed:
You can splash yourself with reagent glasses now

Ikarrus committed:
Adds a thrownby var that's used by admin logging.

TheVekter commented:
Nanotrasen needs OSHA now I guess, huh?

Lumipharon commented:
Essential feature that needs to be added to this.

Toxin special, the drink that uses plasma, and is on fire. If you throw it at someone it should set them on fire for maximum HONK.

Ikarrus committed:
Removed atom var and instead used paramaters to determine the thrower
